### PR TITLE
Composite pattern

### PR DESCRIPTION
**Pull request by VardanyanAlex**

### CODE REVIEW

File: composite/drawables.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+
+#include <iostream>
+
+#include "drawables.h"
+
+namespace DesignPatterns
+{
+
+CRectangle::CRectangle(int x, int y)
```
Comment (EdoHarutyunyan) [inline composite/drawables.cpp:9]:
```suggestion
CRectangle::CRectangle(const int x, const int y)
```

File: composite/drawables.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    : m_x{ x }
+    , m_y{ y }
+{}
+
+void CRectangle::Draw() const
+{
+    std::cout << "Was drawn rectangle with x = " << m_x << "(width) and y = " << m_y << "(height)" << std::endl;
+}    
+
+CLine::CLine(int x, int y)
```
Comment (EdoHarutyunyan) [inline composite/drawables.cpp:19]:
```suggestion
CLine::CLine(const int x, const int y)
```

File: composite/drawables.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+class CLine : public IGraphic
+{
+public:
+    explicit CLine(int x, int y);
+
+    void Draw() const override;
+
+private:
+    int m_x{};
```
Comment (EdoHarutyunyan) [inline composite/drawables.h]: Is it zero after `{}`?

**Review by EdoHarutyunyan — APPROVED** [review]

File: composite/drawables.h (hunk 3 above)
Comment (VardanyanAlex) [reply composite/drawables.h]: Yes, this is the same as to write
int m_x = 0;

from cppreference:
2) Through a default member initializer, which is a brace or equals initializer included in the member declaration and is used if the member is omitted from the member initializer list of a constructor.

**Pull request merged**